server/aws-lsp-identity/src/language-server/profiles/sharedConfigProfileStore.ts

```diff
@@ -81,72 +78,83 @@ export class SharedConfigProfileStore implements ProfileStore {
+        if (data.profiles) {
+            this.applySectionsToParsedIni(
+                IniSectionType.PROFILE,
+                data.profiles,
+                parsedKnownFiles,
+                (section, parsedSection) =>
+                    !section.kinds.includes(ProfileKind.SsoTokenProfile) ||
+                    profileDuckTypers.SsoTokenProfile.eval(parsedSection)
+            )
         }
 
-        this.applySectionsToParsedIni(data.ssoSessions, parsedKnownFiles, section =>
-            ssoSessionDuckTyper.eval(section.settings)
-        )
+        if (data.ssoSessions) {
+            this.applySectionsToParsedIni(
+                IniSectionType.SSO_SESSION,
+                data.ssoSessions,
+                parsedKnownFiles,
+                (_, parsedSection) => ssoSessionDuckTyper.eval(parsedSection)
+            )
+        }
 
         await saveKnownFiles(parsedKnownFiles, init)
     }
 
     private applySectionsToParsedIni<T extends Section>(
+        sectionType: IniSectionType,
         sections: T[],
         parsedKnownFiles: ParsedIniData,
-        validator: (section: T) => boolean
+        validator: (section: T, parsedSection: IniSection) => boolean
     ): void {
         for (const section of sections) {
             if (!section?.name) {
                 throw new Error('Section name is required.')
             }
 
-            if (!validator(section)) {
-                throw new Error(`Section [${section.name}] is invalid.`)
+            const parsedSectionName = new SectionHeader(section.name, sectionType).toParsedSectionName()
+
+            // Remove sections that have no settings
+            if (!section.settings) {
+                delete parsedKnownFiles[parsedSectionName]
+                continue
             }
 
-            const parsedSection = (parsedKnownFiles[section.name] ||= {})
+            const parsedSection = (parsedKnownFiles[parsedSectionName] ||= {})
 
             // eslint-disable-next-line prefer-const
             for (let [name, value] of Object.entries(section.settings)) {
                 if (Array.isArray(value)) {
                     value = normalizeSettingList(value)?.join(',') ?? ''
                 }
 
-                // If and when needed in the future, handle object types for subsections
+                // If and when needed in the future, handle object types for subsections (e.g. api_versions)
 
-                if (!value) {
-                    throw new Error(`Setting [${name}] must have a value.`)
+                // If setting passed with null or undefined then remove setting
+                // If setting passed with any other value then update setting
+                // If setting not passed then preserve setting in file as-is
+                if (value === null || value === undefined) {
+                    Object.hasOwn(parsedSection, name) && delete parsedSection[name]
+                } else {
+                    parsedSection[name] = value.toString()
```

Comment:
When `value` is empty string, it will go to this codepath, is it expected?

Reply:
Good catch.  I wanted to allow falsy values such as 0 and false here, but an empty/whitespace-only string is the same as no value.  Updated here and above on the settings property check to handle these additional no value cases and updated unit tests for these cases.